engines/gob/inter_v6.cpp

```diff
@@ -297,6 +297,26 @@ void Inter_v6::o6_assign(OpFuncParams &params) {
 	} else
 		loopCount = 1;
 
+	// WORKAROUND for a bug in Adibou 2 scripts for cooking activity: bananas count is not updated correctly.
+	// The banana balance equation should be: "remaining bananas = previous remaining bananas - bananas used for cake"
+	// but scripts do instead "remaining bananas = previous remaining *cherries* - bananas used for cake" :p
+	if (_vm->getGameType() == kGameTypeAdibou2
+		&&
+		loopCount == 1
+		&&
+		_vm->_game->_script->pos() == 18631 // same offset in all versions
+		&&
+		(dest == 40956 // bananas in v2.10, v2.11
```

Comment:
May it potentially happen that we modify something else with this `dest` in other games? If it could be the case, then you may add some game flag (GF_) to distinguish between different versions.

Reply:
do we have something here like the enhancement flag like for scumm, too?

Reply:
> May it potentially happen that we modify something else with this dest in other games? If it could be the case, then you may add some game flag (GF_) to distinguish between different versions.

Theoretically, yes, although I think we would be quite unlucky to have the same (opcode, dest, script offset) combination for something else ; but it would be cleaner indeed to activate this workaround only on confirmed versions (the bug it solves is not very serious anyway).